MAINTAINERS_GUIDE.md

```diff
@@ -0,0 +1,90 @@
+And so on.
+
+All decisions affecting this project, big and small, follow the same 3 steps:
+
+* Step 1: Open a pull request. Anyone can do this.
+
+* Step 2: Discuss the pull request. Anyone can do this.
+
+* Step 3: Accept (`LGTM`) or refuse a pull request. The relevant maintainers do this (see below "Who decides what?")
+
+*I'm a maintainer, should I make pull requests too?*
+
+Yes.
+Nobody should ever push to the `main` branch directly.
+All changes should be made through a pull request.
+
+Bigger changes (that may span projects): an [OCI Working Group](https://github.com/opencontainers/tob/blob/main/CHARTER.md) may be needed.
+
+## Who decides what?
+
+All decisions are pull requests, and the relevant maintainers make decisions by accepting or refusing the pull request.
+Review and acceptance by anyone is denoted by adding a comment in the pull request: `LGTM`.
+However, only currently listed `MAINTAINERS` are counted towards the required **two LGTMs**.
+
+Overall the maintainer system works because of mutual respect across the maintainers of the project.
+The maintainers trust one another to make decisions in the best interests of the project.
+Sometimes maintainers can disagree and this is part of a healthy project to represent the point of views of various people.
+
+The maintainer system is built on trust, if there is a conflict the maintainers do not agree on it can be brought to the [technical oversight board](https://github.com/opencontainers/tob) for resolution.
+It is expected that this would be a very exceptional event.
+
+## How are maintainers added?
+
+The best maintainers have a vested interest in the project.
+Maintainers are first and foremost contributors that have shown they are committed to the long term success of the project.
+Contributors wanting to become maintainers are expected to be deeply involved in contributing code, pull request review, and triage of issues in the project for more than two months.
+
+Just contributing does not make you a maintainer, it is about building trust with the current maintainers of the project and being a person that they can depend on and trust to make decisions in the best interest of the project.
+The final vote to add a new maintainer should be approved by over **66% (2/3) of the current maintainers**.
+The voting period is **five business days** on the Pull Request to add the new maintainer.
+
+## What is expected of maintainers?
+
+Part of a healthy project is to have active maintainers to support the community in contributions and perform tasks to keep the project running.
+Maintainers are expected to be able to respond in a timely manner if their help is required on specific issues where they are pinged.
+Being a maintainer is a time consuming commitment and should not be taken lightly.
+
+When a maintainer is unable to perform the required duties they can be removed with a vote by **66% (2/3) of the current maintainers**.
```

Comment:
Consider changing this from "current" to "voting" so that if maintainers do not vote within the 10 day period, they are not counted in the 2/3 majority threshold. This would better prepare the project if several maintainers go inactive and we cannot get 2/3 of all maintainers to respond to a vote.

We should also allow a maintainer to step down themselves, without a vote.

Reply:
The quorum problem is a weird one to solve generically... but maybe to mitigate the "multiple inactive maintainers" scenario we could introduce some requirement around activity/responsiveness? if someone doesn't respond for 6 months they're removed as a maintainer? not sure

Reply:
right. "Active" is PR review, issue comments or creation, slack/mailing-list participation, joining calls?
Seems tricksy to track without a bot. 

so if a maintainer is deemed inactive, then their vote counts for less? or they can be remove with less than 2/3 quorum?

Reply:
My personal view is that the quorum rules should be the same for all governance votes because otherwise you may end up with weird cases where we could rule-lawyer ourselves into weird situations.

The rest of the OCI (both projects and the TOB) use qualified super-majority votes (2/3 of all seats, with abstaining seats counted as being against the motion) so I don't feel particularly comfortable with changing the rules for this one governance procedure in one project. A project with healthy maintainership that has a single maintainer they wish to remove should not have any issues getting a 2/3 vote if that is what the will of the project is.

If we are continuing to have inactive maintainer issues such that such a mechanism cannot work, then we should solve it by removing inactive maintainers (I am aware that I would probably be included under that label as I haven't been involved in the more recent image signing and related work) and being more assertive about it than last time.

But that's just my 2 cents.

Reply:
Would it be useful to have a state between active and removed? Allow a maintainer to be marked as inactive if they aren't responding for X months, so that 2/3rd majorities aren't impossible to achieve. But those inactive maintainers could be easily reactivated if they later returned. Perhaps a 2 active maintainer vote to move someone between active and inactive, but 2/3rds of active maintainers to remove a maintainer or perform other large changes.

I think a policy to handle these situations would be useful in lots of open source governance docs, not just in image-spec. People change jobs or have major life changes that result in us going quiet. And if routine updates still happen with the existing moderators, no one notices until a major change is needed.